gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/PublisherApi.java

```diff
@@ -0,0 +1,498 @@
+    return listTopicSubscriptionsCallable().pageStreaming(LIST_TOPIC_SUBSCRIPTIONS_PAGE_DESC);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Lists the name of the subscriptions for this topic.
+   */
+  public ApiCallable<ListTopicSubscriptionsRequest, ListTopicSubscriptionsResponse> listTopicSubscriptionsCallable() {
+    return ApiUtils.prepareIdempotentCallable(LIST_TOPIC_SUBSCRIPTIONS, settings).bind(channel);
+  }
+
+  // ----- deleteTopic -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes the topic with the given name. Returns NOT_FOUND if the topic does
+   * not exist. After a topic is deleted, a new topic may be created with the
+   * same name; this is an entirely new topic with none of the old
+   * configuration or subscriptions. Existing subscriptions to this topic are
+   * not deleted, but their `topic` field is set to `_deleted-topic_`.
+   *
+   * @param topic Name of the topic to delete.
+   */
+  public void deleteTopic(String topic) {
+    DeleteTopicRequest request =
+        DeleteTopicRequest.newBuilder()
+        .setTopic(topic)
+        .build();
+
+    deleteTopic(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes the topic with the given name. Returns NOT_FOUND if the topic does
+   * not exist. After a topic is deleted, a new topic may be created with the
+   * same name; this is an entirely new topic with none of the old
+   * configuration or subscriptions. Existing subscriptions to this topic are
+   * not deleted, but their `topic` field is set to `_deleted-topic_`.
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public void deleteTopic(DeleteTopicRequest request) {
+    deleteTopicCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes the topic with the given name. Returns NOT_FOUND if the topic does
+   * not exist. After a topic is deleted, a new topic may be created with the
+   * same name; this is an entirely new topic with none of the old
+   * configuration or subscriptions. Existing subscriptions to this topic are
+   * not deleted, but their `topic` field is set to `_deleted-topic_`.
+   */
+  public ApiCallable<DeleteTopicRequest, Empty> deleteTopicCallable() {
+    return ApiUtils.prepareIdempotentCallable(DELETE_TOPIC, settings).bind(channel);
+  }
+
+
+  // ========
+  // Cleanup
+  // ========
+
+  /**
+   * Initiates an orderly shutdown in which preexisting calls continue but new calls are immediately
+   * cancelled.
+   */
+  @Override public void close() {
+    // Manually-added shutdown code
+
+    // Auto-generated shutdown code
+    channel.shutdown();
```

Comment:
Does it make sense that you will shutdown the channel when provided by the user?